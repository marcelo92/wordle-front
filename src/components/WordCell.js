function WordCell(props) {
    let classnames = 'word-cell'
    if (props.check) {
      let validate = ' gray';
      if (props.word[props.index] === props.value) {
        validate = 'red';
      } else if (props.word.includes(props.value)) {
        validate = 'blue'
      }
      classnames += ' ' + validate
    }
    return (<td className={classnames}>{props.value}</td>)
  }

  export default WordCell;