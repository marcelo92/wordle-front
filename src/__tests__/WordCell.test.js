import renderer from 'react-test-renderer';
import WordCell from '../WordCell';

let wordCellProps = {}

beforeEach(() => {
    wordCellProps = {
        value: '',
        word: "irate",
        check: true,
        index: 0
    }
});
  

it('Return <td> with no validation class', () => {
    wordCellProps.check = false;
    const component = renderer.create(<WordCell {...wordCellProps} />);
    let td = component.toJSON();
    expect(td).toMatchSnapshot()
});

it('Return <td> with gray classname for no match', () => {
    wordCellProps.value = 'f';
    const component = renderer.create(<WordCell {...wordCellProps} />);
    let td = component.toJSON();
    expect(td).toMatchSnapshot()
});

it('Return <td> with blue classname for letter in wrong index', () => {
    wordCellProps.value = 'e';
    const component = renderer.create(<WordCell {...wordCellProps} />);
    let td = component.toJSON();
    expect(td).toMatchSnapshot()
});

it('Return <td> with red classname for letter in wrong index', () => {
    wordCellProps.value = 'i';
    const component = renderer.create(<WordCell {...wordCellProps} />);
    let td = component.toJSON();
    expect(td).toMatchSnapshot()
});