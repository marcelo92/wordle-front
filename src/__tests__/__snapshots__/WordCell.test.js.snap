// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Return <td> with blue classname for letter in wrong index 1`] = `
<td
  className="word-cell blue"
>
  e
</td>
`;

exports[`Return <td> with gray classname for no match 1`] = `
<td
  className="word-cell  gray"
>
  f
</td>
`;

exports[`Return <td> with no validation class 1`] = `
<td
  className="word-cell"
/>
`;

exports[`Return <td> with red classname for letter in wrong index 1`] = `
<td
  className="word-cell red"
>
  i
</td>
`;
